src/main/scala/com/databricks/labs/overwatch/validation/DataReconciliation.scala

```diff
@@ -0,0 +1,327 @@
+                                  ):Array[ReconReport]={
+    spark.conf.set("spark.sql.legacy.allowHashOnMapType","true")
+      val reconStatus: ArrayBuffer[ReconReport] = new ArrayBuffer[ReconReport]()
+      sourceOrgIDArr.foreach(orgId=> {
+        targets.foreach(target => {
+          reconStatus.append(hashValidation(target, orgId, sourceEtl, targetEtl))
+        })
+      }
+      )
+    spark.conf.set("spark.sql.legacy.allowHashOnMapType","false")
+    reconStatus.toArray
+  }
+
+  /**
+   * Function saves the recon report.
+   * @param reconStatusArray
+   * @param path
+   * @param reportName
+   * @param reconRunId
+   */
+  private def saveReconReport(reconStatusArray: Array[ReconReport], path: String, reportName: String, reconRunId: String): Unit = {
+    val validationPath = {
+      if (!path.startsWith("dbfs:") && !path.startsWith("s3") && !path.startsWith("abfss") && !path.startsWith("gs")) {
+         s"""dbfs:${path}"""
+      }else{
+        path
+      }
+    }
+
+    val pipelineSnapTime =  Pipeline.createTimeDetail(LocalDateTime.now(Pipeline.systemZoneId).toInstant(Pipeline.systemZoneOffset).toEpochMilli)
+    reconStatusArray.toSeq.toDS().toDF()
+      .withColumn("reconRunId", lit(reconRunId))
+      .withColumn("snapTS", lit(pipelineSnapTime.asTSString))
+      .withColumn("timestamp", lit(pipelineSnapTime.asUnixTimeMilli))
+      .write.format("delta")
+      .option("mergeSchema", "true")
+      .mode("append")
+      .save(s"""${validationPath}/report/${reportName}""")
+    println("ReconRunID:"+reconRunId)
+    println("Validation report has been saved to " + s"""${validationPath}/report/${reportName}""")
+  }
+
+
+  /**
+   * This method generates a query to fetch data from a table for a specific organization.
+   *
+   * @param tableName : Name of the table
+   * @param orgId     : Organization ID
+   * @return String : Query
+   */
+  private def getQuery(tableName: String, orgId: String): String = {
+    s"""select * from $tableName where organization_id = ${orgId} """
+  }
+
+
+  /**
+   * This method fetches a DataFrame from a table using a query.
+   *
+   * @param query  : Query to fetch data
+   * @param target : PipelineTable object
+   * @return DataFrame
+   */
+  private def getTableDF(query: String,target: PipelineTable):DataFrame = {
+    try{
+      val excludedCol = target.excludedReconColumn
+      val dropCol = excludedCol ++ Array("Overwatch_RunID", "Pipeline_SnapTS", "__overwatch_ctrl_noise")
+      val filterDF = spark.sql(query).drop(dropCol: _ *)
+      filterDF
+    }catch {
+      case exception: Exception =>
+        println(s"""Exception: Unable to run the query ${query}"""+exception.getMessage)
+        spark.emptyDataFrame
+    }
+
+  }
+
+  /**
+   * This method fetches all targets for a workspace.
+   *
+   * @param workspace : Workspace object
+   * @return ParArray of PipelineTable
+   */
+  private[overwatch] def getAllTargets(workspace: Workspace): ParArray[PipelineTable] = {
```

Comment:
this function is heavily used during our cloning process, might be worth it to put it in Helpers so that we can use this instead